clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredLoginCallbackHandler.java

```diff
@@ -190,6 +204,29 @@ private void handleCallback(OAuthBearerTokenCallback callback) throws IOExceptio
         }
     }
 
+    /**
+     *  Add and validate all the configured extensions.
+     *  Token keys, apart from passing regex validation, must not be equal to the reserved key {@code OAuthBearerClientInitialResponse.AUTH_KEY}
+     */
+    private void handleExtensionsCallback(SaslExtensionsCallback callback) {
+        Map<String, String> extensions = new HashMap<>();
+        for (Map.Entry<String, String> configEntry : this.moduleOptions.entrySet()) {
+            String key = configEntry.getKey();
+            if (!key.startsWith(EXTENSION_PREFIX))
+                continue;
+            String extensionName = key.substring(EXTENSION_PREFIX.length());
+            String extensionValue = configEntry.getValue();
+
+            if (!OAuthBearerClientInitialResponse.EXTENSION_KEY_PATTERN.matcher(extensionName).matches() || key.equals(OAuthBearerClientInitialResponse.AUTH_KEY))
+                throw new ConfigException("Extension name " + extensionName + " is invalid");
+            if (!OAuthBearerClientInitialResponse.EXTENSION_VALUE_PATTERN.matcher(extensionValue).matches())
+                throw new ConfigException("Extension value (" + extensionValue + ") for extension " + extensionName + " is invalid");
```

Comment:
Looks like duplicate code. Couldn't we just use one static `OAuthBearerClientInitialResponse.validateExtensions(Map<String, String>)` method for validation?

Reply:
Yes we can. This will result in a bit more complicated code in `OAuthBearerUnsecuredLoginCallbackHandler#handleExtensionsCallback()` since it needs to unprefix the extensions first